simulator/func_sim/func_sim.cpp

```diff
@@ -18,12 +18,9 @@ MIPS::~MIPS()
 
 std::string MIPS::step()
 {
-    // fetch
-    uint32 instr_bytes = mem->fetch( PC);
-
-    // decode
-    FuncInstr instr( instr_bytes, PC);
-
+    // fetch instruction
+    FuncInstr instr = mem -> fetch_instr( PC);
```

Comment:
We do not add spaces around "->" according to our code style guidelines: https://github.com/MIPT-ILab/MIPT-Vis/wiki/Syntax